sdb/commands/zfs/range_tree.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+from typing import Iterable
+
+import drgn
+import sdb
+from sdb.commands.zfs.btree import Btree
+from sdb.commands.cast import Cast
+
+
+class RangeTree(sdb.PrettyPrinter):
+    names = ['range_tree']
+    input_type = 'range_tree_t *'
+
+    def pretty_print(self, objs: Iterable[drgn.Object]) -> None:
+
+        class RangeTreeSeg(sdb.Command):
+
+            def __init__(self, rt: drgn.Object):
+                super().__init__()
+                self.rt = rt
+
+            def call(self, objs: Iterable[drgn.Object]) -> None:
+                for seg in objs:
+                    start = (
+                        seg.rs_start << self.rt.rt_shift) + self.rt.rt_start
+                    end = (seg.rs_end << self.rt.rt_shift) + self.rt.rt_start
+                    if hasattr(self.rt, 'rs_fill'):
+                        fill = seg.rs_fill << self.rt.rt_shift
+                        print(f"    [{hex(start)} {hex(end)}) "
+                              f"(length {hex(end - start)}) "
+                              f"(fill {hex(fill)})")
+                    else:
+                        print(f"    [{hex(start)} {hex(end)}) "
+                              f"(length {hex(end - start)})")
+
+        for rt in objs:
+            print(f"{hex(rt)}: range tree of {int(rt.rt_root.bt_num_elems)} "
+                  f"entries, {int(rt.rt_space)} bytes")
+
+            for _ in sdb.execute_pipeline([rt], [RangeSeg(), RangeTreeSeg(rt)]):
+                pass
+
+
+class RangeSeg(sdb.Locator):
+    names = ['range_seg']
+
+    #pylint: disable=no-self-use
```

Comment:
What is this warning for?

Reply:
This method doesn't use `self`, so it could be rewritten as a `@staticmethod`.  However, I don't think that would work because the `@sdb.InputHandler('range_tree_t *')` annotation assumes it's on an instance method. (at least, we find it with `inspect.getmembers(self, inspect.ismethod)`).